### PR TITLE
Feat/#1095 get single program entity exception

### PR DESCRIPTION
**Pull request by ciaranschutte**
## Link to Issue
https://github.com/icgc-argo/roadmap/issues/1092

### Type of Change

- [ ] Bug
- [ ] Refactor
- [ ] New Feature
- [ ] Release Candidate

### Checklist before requesting review:

- [ ] Check branch (code change PRs go to `develop` not master)
- [ ] Check copyrights for new files
- [ ] Manual testing
- [ ] Regression tests completed and passing (double check number of tests).
- [ ] Spelling has been checked.
- [ ] Updated swagger docs accordingly (check it's still valid)
- [ ] Set `validationDependency` in meta tag for [Argo Dictionary](https://github.com/icgc-argo/argo-dictionary) fields used in code

### CODE REVIEW

File: src/exception/missing-entity-exceptions/api.ts [hunk 1]
```diff
@@ -22,9 +22,20 @@ import { Request, Response } from 'express';
...
+import { getByProgramId, listAll } from './repo';
 
 class MissingEntityExceptionController {
+	@HasFullReadAccess()
+	async getProgramException(req: Request, res: Response) {
+		const result = await getByProgramId(req.params.programId);
+		if (!result.success) {
+			const { message, errors } = result;
+			return res.status(500).send({ message, errors });
+		} else {
```
Comment (joneubank) [inline src/exception/missing-entity-exceptions/api.ts:34]: When we have no database entry, can we instead reply with a blank exception object?
example:
```ts
{
  programId: "SomeId",
  donorSubmitterIds: []
}
```

Comment (joneubank) [reply src/exception/missing-entity-exceptions/api.ts:34]: Also note for reference... if the response shows no entry for the parameters the user requested, thats a 4xx error of some sort, based on user inputs, not a 5xx server error.

**Review by joneubank — CHANGES_REQUESTED** [review]
Requesting change to response when we have no DB record for a program ID.

File: src/exception/missing-entity-exceptions/api.ts (hunk 1 above)
Comment (ciaranschutte) [reply src/exception/missing-entity-exceptions/api.ts:34]: If there's no db entry that's happening already https://github.com/icgc-argo/argo-clinical/blob/feat%2Fmissing-entity-exception/src/exception/missing-entity-exceptions/repo.ts#L84

Comment (ciaranschutte) [reply src/exception/missing-entity-exceptions/api.ts:34]: in this case though it is sent with status 200 and the object you suggested above

**Review by joneubank — APPROVED** [review]

File: src/exception/missing-entity-exceptions/api.ts (hunk 1 above)
Comment (joneubank) [reply src/exception/missing-entity-exceptions/api.ts:34]: Yes that is correct lol. Already handled.

**Pull request merged**